lib/formtastic/helpers/inputs_helper.rb

```diff
@@ -350,12 +350,21 @@ def association_columns(*by_associations) # @private
         end
       end
 
+      # Collects non-standard foreign key columns
+      def foreign_key_columns
+        if @object.present? && @object.class.respond_to?(:reflections)
+          @object.class.reflections.map {|_,v| v.options[:foreign_key] }.compact
```

Comment:
@mikz, 

>> What if rails changes format of that object or that method? Because it is a mock no one can know because it does not test the rails interface.


generally I agree with you about test changes,  however  formtastic tests already over-mocked and over-stubbed and needs more efforts and time to add test for each edge case than it deserves.

>>  I'll start a branch soon exploring that and replacing few mocks with ActiveRecord objects.

feel free to ping me when it will be more clear how to arrange tests after this, so I could finish it if I have free time.